modules/folder/variables.tf

```diff
@@ -14,6 +14,27 @@
  * limitations under the License.
  */
 
+variable "assured_workload_config" {
+  description = "Create AssuredWorkloads folder instead of regular folder when value is provided. Incompatible with folder_create=false."
+  type = object({
+    compliance_regime         = string
```

Comment:
Should this be validated as an enum?

Reply:
The list of values is quite big and most important it's growing over time, now it would be something like:
```
  validation {
    condition     = contains([
      "COMPLIANCE_REGIME_UNSPECIFIED",
      "IL4",
      "CJIS",
      "FEDRAMP_HIGH",
      "FEDRAMP_MODERATE",
      "US_REGIONAL_ACCESS",
      "HIPAA, HITRUST",
      "EU_REGIONS_AND_SUPPORT",
      "CA_REGIONS_AND_SUPPORT",
      "ITAR",
      "AU_REGIONS_AND_US_SUPPORT",
      "ASSURED_WORKLOADS_FOR_PARTNERS",
      "ISR_REGIONS",
      "ISR_REGIONS_AND_SUPPORT",
      "CA_PROTECTED_B, IL5",
      "IL2", 
      "JP_REGIONS_AND_SUPPORT"
      ], var.assured_workload_config.compliance_regime)
    error_message = "Allowed values for input_parameter are: COMPLIANCE_REGIME_UNSPECIFIED, IL4, CJIS, FEDRAMP_HIGH, FEDRAMP_MODERATE, US_REGIONAL_ACCESS, HIPAA, HITRUST, EU_REGIONS_AND_SUPPORT, CA_REGIONS_AND_SUPPORT, ITAR, AU_REGIONS_AND_US_SUPPORT, ASSURED_WORKLOADS_FOR_PARTNERS, ISR_REGIONS, ISR_REGIONS_AND_SUPPORT, CA_PROTECTED_B, IL5, IL2, JP_REGIONS_AND_SUPPORT."
  }
```
Should I still go for it? @sruffilli @ludoo

Reply:
LGTM
If you want to make it a bit shorter, you can send the user to the docs instead of listing the values again: "Field compliance_regime must be one of the values listed in https://cloud.google.com/assured-workloads/docs/reference/rest/Shared.Types/ComplianceRegime."

Reply:
I would say yes, and just remove the list from the error message, saying something like "check the variable validation or resource documentation for values".

Reply:
Done